### ISSUE TITLE
broken input when cursor is on second to last character

### ISSUE THREAD

**Issue by JeffreyHofmann** [piotrmurach/tty-prompt#94]
This is my first time ever submitting an issue request, I hope I've described it well enough
### Describe the problem
when the cursor is right before the end of your input string, when you type the characters are placed at the last position rather than the second to last position
only time it works correctly is when string is length 1

### Steps to reproduce the problem
tty.ask, type any character, move the cursor back, and type the next character twice
```
#!/usr/bin/env ruby
require 'tty-prompt'
prompt = TTY::Prompt.new(interrupt: lambda {exit})
puts input = prompt.ask('broken input?')
```

### Actual behaviour
when typing a (arrow key left) bb, you're left with bab

### Expected behaviour
it should be bba
![image](https://user-images.githubusercontent.com/25801559/48242419-208eae80-e3a9-11e8-8f00-8c6d10b17064.png)

### Env
ruby -v
ruby 2.3.7p456

**Comment by piotrmurach** [comment]
Hi Jeffrey! 👋 

Thanks for using the library and submitting this great issue report.

The input processing is handled by another gem [tty-reader](https://github.com/piotrmurach/tty-reader) and this is the place where I would start looking for the bug. Would you have time to investigate? Or even submit your first PR? The codebase is fairly small and there is a line representation that maybe at fault, that's where I would start looking anyway.

**Comment by JeffreyHofmann** [comment]
HI! I would be open to submitting a PR, however I don't totally follow the logic creating the problem, so I don't want to fix it and upset something else. I know exactly where the issue is occurring, I've commented it in the screenshot. I have 0 idea what the case i've highlighted would be useful for. 
![screen shot 2018-11-15 at 2 28 00 am](https://user-images.githubusercontent.com/25801559/48536969-17588280-e87e-11e8-8e0f-74befb99ce77.png)

As a side note, is there any way to access the contents of the line object while asking something with a tty prompt? Don't see any way to do it with the reader so I ended up using print statements in the line source. If not, have you considered making it accessible through the reader?

**Comment by piotrmurach** [comment]
This was super useful! I went ahead and fixed the [issue](https://github.com/piotrmurach/tty-reader/commit/cdec2cc1ae25b0d6b4eedcd26d0546e6668bab82) in tty-reader.

Re the line addition, `tty-prompt` is not always line oriented, some prompts work only on keypress such as `mask`. So the line doesn't always exist per say. However, I have changed `tty-reader` to expose current line on the event object:

```ruby
prompt.on(:keypress) do |event|
  puts "Current line: #{event.line}"
end
``` 

So hopefully this will help you in future to debug things.

Latest `v0.18.0` release contains this and other fixes.
